core/domain/draft_upgrade_services.py

```
@@ -343,17 +343,19 @@ def _convert_states_v33_dict_to_v34_dict(cls, draft_change_list):
                             )
             elif (change.property_name ==
                   exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME):
-                new_value = (
-                    state_domain.Outcome.convert_html_in_outcome(
-                        new_value, conversion_fn))
+                if new_value is not None:
```

is it expected to have DEFAULT_OUTCOME to be None? If not should we raise an error?

---

I think it is possible for it to be None, e.g. that makes sense for EndExploration. The existing backend code supports this, e.g. see core/domain/exp_domain.py in Exploration.from_dict() and also validate().